src/queue.js

```diff
@@ -204,4 +204,50 @@ Queue.prototype.shutdown = function() {
   }));
 };
 
+/**
+ * Gets queue worker count
+ */
+Queue.prototype.getWorkerCount = function() {
+  return this.workers.length;
+};
+
+/**
+ * Adds a queue worker
+ */
+Queue.prototype.addWorker = function() {
+  logger.debug('Queue: adding worker');
+  var processId = (this.specId ? this.specId + ':' : '') + this.workers.length;
+  var worker = new QueueWorker(
+    this.tasksRef,
+    processId,
+    this.sanitize,
+    this.suppressStack,
+    this.processingFunction
+  );
+  this.workers.push(worker);
+
+  if (_.isUndefined(this.specId)) {
+    worker.setTaskSpec(DEFAULT_TASK_SPEC);
+  // if the currentTaskSpec is not yet set it will be called once it's fetched
+  } else if (this.currentTaskSpec) {
+    worker.setTaskSpec(this.currentTaskSpec);
+  }
+  return worker;
+};
+
+/**
+ * Removes a queue worker if one exists
+ */
+Queue.prototype.shutdownWorker = function() {
+  logger.debug('Queue: removing worker');
+  var shutdownPromise;
+  var worker = this.workers.shift();
+
+  if (worker) {
+    shutdownPromise = worker.shutdown();
+  }
+  return shutdownPromise;
```

Comment:
d. return `RSVP.reject(new Error('No workers to shutdown'))` or similar

I think it should be a promise rather than `undefined` or a synchronous error, and I can see an argument that this is a failure case - I'm leaning towards d.